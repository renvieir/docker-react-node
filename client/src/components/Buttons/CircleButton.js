import React from 'react';
import './CircleButton.css';

export default function CircleButton(props) {
  return (
    <button
      className={"button--circle button--image " + (props.minus ? "image--minus": "image--plus")}
      onClick={props.handleClick}
      disabled={props.disabled}
    >
    </button>
  );
}
