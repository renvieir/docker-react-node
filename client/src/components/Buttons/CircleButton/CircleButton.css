button:hover {
  cursor: pointer;
}

.button--circle {
  border-radius: 50%;
  border: 1px solid royalblue;
  background-color: white;
  width: 24px;
  height: 24px;
  margin: auto;
}

.button--circle[disabled] {
  border-color: #d2d2d2;
  background-color: lightgray;
  color: grey;
  box-shadow: none;
}

.button--image {
  background-size: 12px;
  background-repeat: no-repeat;
  background-position: center;
  color: black;
}

.image--minus {
  background-image: url(./icons-minus.png);
}

.image--plus {
  background-image: url(./icons-plus.png);
}
