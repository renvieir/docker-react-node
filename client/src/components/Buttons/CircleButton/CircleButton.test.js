import React from 'react';
import { shallow } from 'enzyme';

import toJson from 'enzyme-to-json';
import CircleButton from './CircleButton';

describe('CircleButton', () => {
  let wrapper;
  const mocks = {};

  beforeEach(() => {
    wrapper = null;

    mocks.props = {
      disabled: false,
      handleClick: jest.fn(),
    }
  });

  it('should render plus icon as default', () => {
    wrapper = shallow(<CircleButton {...mocks.props}  />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render minus icon', () => {
    mocks.props.minus = true;
    wrapper = shallow(<CircleButton {...mocks.props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render disabled', () => {
    mocks.props.disabled = true;
    wrapper = shallow(<CircleButton {...mocks.props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleCLick', () => {
    wrapper = shallow(<CircleButton {...mocks.props} />);
    wrapper.find('button').simulate('click');
    expect(mocks.props.handleClick).toBeCalled();
  });
});