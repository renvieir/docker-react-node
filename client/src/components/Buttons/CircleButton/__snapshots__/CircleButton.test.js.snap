// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleButton should render disabled 1`] = `
<button
  className="button--circle button--image image--plus"
  disabled={true}
  onClick={[MockFunction]}
/>
`;

exports[`CircleButton should render minus icon 1`] = `
<button
  className="button--circle button--image image--minus"
  disabled={false}
  onClick={[MockFunction]}
/>
`;

exports[`CircleButton should render plus icon as default 1`] = `
<button
  className="button--circle button--image image--plus"
  disabled={false}
  onClick={[MockFunction]}
/>
`;
